packages/dev/core/src/Engines/thinEngine.ts

```diff
@@ -3055,10 +3072,8 @@ export class ThinEngine extends AbstractEngine {
      * @returns The options to pass to texImage2D or texImage3D calls.
      * @internal
      */
-    public _getTexImageParametersForCreateTexture(babylonFormat: Nullable<number>, fileExtension: string, useSRGBBuffer: boolean): TexImageParameters {
-        if (babylonFormat === undefined || babylonFormat === null) {
-            babylonFormat = fileExtension === ".jpg" && !useSRGBBuffer ? Constants.TEXTUREFORMAT_RGB : Constants.TEXTUREFORMAT_RGBA;
-        }
+    public _getTexImageParametersForCreateTexture(babylonFormat: Nullable<number>, useSRGBBuffer: boolean): TexImageParameters {
+        babylonFormat = babylonFormat ?? Constants.TEXTUREFORMAT_UNDEFINED;

```

Comment:
Can we remove nullable on the format and not fall back to undefined here ? I agree the api let undefined come through but as it is only ever called in one place I d rather force the check in the caller ?

Reply:
Yes, we could either:
1. Keep the same exact logic, where the single caller just passes in `texture.format ?? Constants.TEXTUREFORMAT_UNDEFINED`.
2. Have the caller throw an error if `texture.format` is undefined at that point (it never should be since it is now assigned earlier in the flow).

Preference on these options or a different one?

Reply:
Actually, texture.format is just a number, it is not Nullable. It technically could be -1 if there was some future other caller of _getTexImageParametersForCreateTexture, but then the conditions inside this function would fall through to defaults anyway. So I think checking the format and/or defaulting it is not actually needed anymore.